core/psh/psh.c

```diff
@@ -805,8 +807,9 @@ static int psh_readcmd(struct termios *orig, psh_hist_t *cmdhist, char **cmd)
 		return err;
 	}
 
-	if (*cmd == NULL)
-		psh_exit(EXIT_SUCCESS);
+	if (*cmd == NULL) {
+		return -ENODEV;
+	}
```

Comment:
```suggestion
	if (*cmd == NULL)
		return -ENODEV;
```
nitpicking: coding style